### ISSUE TITLE
Android plugin failing to load

### ISSUE THREAD

**Issue by neoncube2** [godotengine/godot#63727]
### Godot version

3.4.4-stable

### System information

Windows 11

### Issue description

I've created an Android plugin named `StepsPlugin`. The GoDot editor recognizes this plugin and lets me enable it, but when I try to load it via `Engine.get_singleton("StepsPlugin")`, I get `null`.

### Steps to reproduce

Create an Android plugin, like so:
```java

```

Build the plugin.

Create a `.gdap` file, like so:
```ini
[config]
name="StepsPlugin"
binary_type="local"
binary="StepsPlugin-release.aar"
```

In the Godot editor, enable custom Android builds and enable the plugin:
![steps plugin](https://user-images.githubusercontent.com/10658081/182033905-5cfa1e40-45a2-4e74-b70b-692ea3721355.png)

Add a script to the root node that tries to load `StepsPlugin` and call a method on it:

```
extends Node2D

func _ready():
	print("getting StepsPlugin")
	var stepsPlugin = Engine.get_singleton("StepsPlugin")
```

Export the `.apk`.

In Android Studio, open the project and run it on the Android phone. I get this error:

```
I/godot: getting StepsPlugin
E/godot: ERROR: Failed to retrieve non-existent singleton 'StepsPlugin'.
E/godot:    at: get_singleton_object (core/engine.cpp:202) - Condition "!E" is true. Returned: nullptr
```

### Minimal reproduction project

I'd be happy to provide a repo, if it'd be helpful :)

**Comment by neoncube2** [comment]
If anyone has any suggestions, please let me know :)  I've been stuck on this for several days XD

**Comment by neoncube2** [comment]
@Calinou Any suggestions for how I might go about debugging this? :)

I'm thinking that maybe [Java_org_godotengine_godot_plugin_GodotPlugin_nativeRegisterSingleton](https://github.com/godotengine/godot/blob/b777454097701595cb22a9e9249f76bec0e850d0/platform/android/plugin/godot_plugin_jni.cpp#L44) isn't being called, but I've been unable to figure out where the code is that calls that function. (Any hints? :) )

More generally, does Godot emit any logging messages when it's loading a plugin? I'd love to be able to turn on logging and see a trace of Godot loading plugins :)

**Comment by neoncube2** [comment]
I tried putting a `LOG.d()` statement in the plugin's constructor, but the statement isn't showing up in Logcat, so I'm assuming that the plugin isn't being loaded.

**Comment by Calinou** [comment]
> Any suggestions for how I might go about debugging this? :)

I don't work with Android plugins, sorry.

**Comment by neoncube2** [comment]
@Calinou Ah, okay, no worries! 🙂  I found a plugin that's similar to mine (simple Android plugin with no native libraries), so I guess I should try to compile it and see if it works. If not, I guess there must be an issue with my setup.

Plugin: https://github.com/Lamelynx/GodotGetImagePlugin-Android

**Comment by neoncube2** [comment]
Interesting... the same thing happens when I try to use https://github.com/Lamelynx/GodotGetImagePlugin-Android . I'm able to build it, and Godot sees the plugin and lets me activate it, but when I try to load the plugin as a singleton, I always get `null`.

**Comment by Poldovico** [comment]
I'm having the exact same thing. Enable the plugin in the editor, project builds fine, Engine.get_singleton("MyPlugin") returns null.
What's more, this happens both with 3.5 and with 3.4.4, with my old plugin and with two plugins I created from scratch to test this. The old plugin used to work on 3.4.4, so I'm thinking if something changed with the ecosystem that broke this for Godot.

**Comment by finepointcgi** [comment]
I am also running into this with Godot 4 trying to update the godot android share plugin however with Google play billing plugin it seems fine.

**Comment by finepointcgi** [comment]
After some testing it "may" have something to do with the package path.
I took the 
https://github.com/Shin-NiL/Godot-Android-Share-Plugin/tree/main/share-plugin
plugin and updated to be in the package space from 
package shinnil.godot.plugin.android.godotshare;
to
package org.godotengine.godot.plugin.GodotAndroidShare;
and that seemed to make it work. Ill test with others to see if it fixes it.

**Comment by rhighs** [comment]
Same happens on my machine.

OS: Manjaro Linux x86_64
Engine: 4.1-stable

Built a simple Android plugin like OP, the engine seems to load it just fine and lets me activate it via export setting. However, the following script attached to the root note will print `<Object#null>`:
```gdscript
extends Node2D


# Called when the node enters the scene tree for the first time.
func _ready():
	print(Engine.get_singleton("HelloWorld"))


# Called every frame. 'delta' is the elapsed time since the previous frame.
func _process(delta):
	pass

```

**Comment by rhighs** [comment]
@finepointcgi I tried reproducing what you said, basically just changing the package namespace https://github.com/rhighs/godot-android-simple-plugin

**Comment by neoncube2** [comment]
@rhighs What did you change the package namespace to? `package org.godotengine.godot.plugin.GodotAndroidSimplyPlugin;` ?

**Comment by rhighs** [comment]
@neoncube2 Yes but it didn't work, the only thing I had left to do was running my demo on the device via adb rather than editor and it worked flawlessly. I'd like it to work on the editor as well as the feedback loop gets quite shorter compared to shipping the apk to my device/emulator.
